File: 01-sls-app-basico/getUserspy/handler.py
import json
import os
import boto3
from botocore.exceptions import ClientError

# Configuración del cliente DynamoDB
dynamodb_client = boto3.client("dynamodb", region_name=os.getenv("AWS_REGION"))

def get_users(event, context):
    """
    Obtiene todos los usuarios de la tabla DynamoDB.
    """
    try:
        table_name = os.getenv("TABLA_USUARIOS")
        if not table_name:
            raise ValueError("Environment variable 'TABLA_USUARIOS' is not set.")

        # Escanea la tabla
        response = dynamodb_client.scan(TableName=table_name)
        items = response.get("Items", [])

        # Convertir a un formato legible
        users = [deserialize_dynamodb_item(item) for item in items]

        return {
            "statusCode": 200,
            "body": json.dumps(users)
        }
    except ClientError as error:
        print(f"ClientError: {error}")
        return {
            "statusCode": 500,
            "body": json.dumps({"message": f"Error retrieving users: {error.response['Error']['Message']}"})
        }
    except Exception as error:
        print(f"Error: {error}")
        return {
            "statusCode": 500,
            "body": json.dumps({"message": str(error)})
        }


def deserialize_dynamodb_item(item):
    """
    Convierte un objeto DynamoDB en un diccionario legible.
    """
    from boto3.dynamodb.types import TypeDeserializer

    deserializer = TypeDeserializer()
    return {key: deserializer.deserialize(value) for key, value in item.items()}

File: 01-sls-app-basico/getUsers/handler.js
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClient, ScanCommand } from "@aws-sdk/lib-dynamodb";

let dynamoDBClientParams = { region: process.env.AWS_REGION };

const client = new DynamoDBClient(dynamoDBClientParams);
const docClient = DynamoDBDocumentClient.from(client);

export const get = async (event, context) => {
  try {
    const params = {
      TableName: process.env.TABLA_USUARIOS,
    };

    const result = await docClient.send(new ScanCommand(params));

    return {
      statusCode: 200,
      body: JSON.stringify(result.Items),
    };
  } catch (error) {
    console.error(error);
    return {
      statusCode: 500,
      body: JSON.stringify({ message: error.message }),
    };
  }
};
